site/content/docs/agentic/ray-mcp/index.md

```diff
@@ -0,0 +1,475 @@
+  --from-literal=hf_api_token=$HF_TOKEN \
+  --dry-run=client -o yaml | kubectl apply -f -
+```
+
+## Step 2: Containerize and Deploy the Ray Serve Application
+
+Build and push the container image for the Ray Serve application to Artifact Registry.
+
+Navigate to the ray-serve-vllm directory:
+
+```bash
+cd ../ray-serve-vllm
+```
+
+Build and push the container image:
+
+```bash
+gcloud builds submit \
+    --tag us-docker.pkg.dev/$PROJECT_ID/llama-ray-mcp/llama-ray-serve:latest \
+    --project=$PROJECT_ID \
+    .
+```
+
+Deploy the Ray Serve application:
+
+1. Open the `ray-service.yaml` file in the `ray-serve-vllm` directory.
+2. Replace `<PROJECT_ID>` with your Google Cloud project name.
+3. Install kuberay-operator via Helm:
+
+```bash
+# Add the Helm repo
+helm repo add kuberay https://ray-project.github.io/kuberay-helm/
+helm repo update
+
+# Confirm the repo exists
+helm search repo kuberay --devel
+
+# Install both CRDs and KubeRay operator v1.3.2.
+helm install kuberay-operator kuberay/kuberay-operator --version 1.3.2
+
+# Check the KubeRay operator Pod in `default` namespace
+kubectl get pods
+# NAME                                READY   STATUS    RESTARTS   AGE
+# kuberay-operator-6fcbb94f64-mbfnr   1/1     Running   0          17s
+```
+
+4. Apply the manifest:
+```bash
+kubectl apply -f ray-service.yaml
+```
+
+Verify the deployment:
+
+- Check the Ray service status:
+```bash
+kubectl get rayservice llama-31-8b -o yaml
+```
+
+The `serviceStatus` should be `running`.
+
+- Check the raycluster:
+```bash
+kubectl get raycluster
+```
+
+And you should see output similar to this:
+```log
+NAME                           DESIRED WORKERS   AVAILABLE WORKERS   CPUS   MEMORY   GPUS   STATUS   AGE
+llama-31-8b-raycluster-qgzmk   1                 1                   10     33Gi     2      ready    58m
+```
+
+- Check the pods:
+```bash
+kubectl get pods
+```
+
+You should see two pods: one Ray Head Pod and one Ray Worker Pod where Worker pods run the LLM and allow for scaling as the requests grow in volume and Head pod is used for management and should stay at a single replica.
+```log
+NAME                                                  READY   STATUS    RESTARTS   AGE
+kuberay-operator-bb8d4d9c4-7h2vg                      1/1     Running   0          9m33s
+llama-31-8b-raycluster-w9jzw-gpu-group-worker-9l5zx   1/1     Running   0          9m7s
+llama-31-8b-raycluster-w9jzw-head-45rx4               1/1     Running   0          9m8s
+```
+
+- Check the config map for the chat template:
+
+```bash
+kubectl get configmaps
+```
+
+Which should display output similar to:
+```bash
+NAME                   DATA   AGE
+kube-root-ca.crt       1      4h47m
+llama-chat-templates   1      3h18m
+```
+### Test the Ray Serve Deployment
+
+Test the model by setting up port forwarding and sending requests.
+
+Set up port forwarding to the Ray Serve endpoint:
+
+```bash
+kubectl port-forward service/llama-31-8b-serve-svc 8000:8000
```

Comment:
We should mention that this port forward command has to be terminated, before we do the next port-forwards, since they use same port number.